qa/tasks/vstart_runner.py

```diff
@@ -405,10 +400,41 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
 
         return args
 
+    @staticmethod
+    def split_args(args):
+        # Since Python's shell simulation can only work when commands are
+        # provided as a list of argumensts...
+        if isinstance(args, str):
+            if '\n' in args:
+                args = args.split('\n')
+                args = [a.split() for a in args]
+            else:
+                args = args.split()
+        return args
```

Comment:
i'd suggest normalize the `args` here instead of returning different instances of different types, and let the caller to dispatch by types. this is error prune, and difficult to maintain. could just put
```py
if isinstance(args, str):
    return [cmd.split() for cmd in args.split]
else:
    assert(isinstance(args, list))
    assert(all(isinstance(arg, str) for arg in args))
    return [args]
```

Reply:
Yes, keeping a single return value type would be better. I've made the change.